analyzer/src/main/kotlin/managers/utils/PythonInspector.kt

```diff
@@ -113,3 +137,72 @@ internal class PythonInspectorParty(
     val email: String?,
     val url: String?
 )
+
+private const val TYPE = "PyPI"
+
+internal fun List<PythonInspectorPackage>.toOrtPackages(): SortedSet<Package> =
+    groupBy { "${it.name}:${it.version}" }.mapTo(sortedSetOf()) { (_, packages) ->
+        // The python inspector currently often contains two entries for a package where the only difference is the
+        // download URL. In this case, one package contains the URL of the binary artifact, the other for the source
+        // artifact. So take all metadata from the first package except for the artifacts.
+        val pkg = packages.first()
+
+        fun PythonInspectorPackage.getHash(): Hash =
+            when {
+                sha512 != null -> Hash(sha512, HashAlgorithm.SHA512)
+                sha256 != null -> Hash(sha256, HashAlgorithm.SHA256)
+                sha1 != null -> Hash(sha1, HashAlgorithm.SHA1)
+                md5 != null -> Hash(md5, HashAlgorithm.MD5)
+                else -> Hash.NONE
+            }
+
+        fun getArtifact(fileExtension: String) =
+            packages.find { it.downloadUrl.endsWith(fileExtension) }?.let {
+                RemoteArtifact(
+                    url = it.downloadUrl,
+                    hash = it.getHash()
+                )
+            } ?: RemoteArtifact.EMPTY
+
+        Package(
+            // The package has a namespace property which is currently always empty. Deliberately set the namespace to
+            // an empty string here to be consistent with the resolved packages which do not have a namespace property.
+            id = Identifier(type = TYPE, namespace = "", name = pkg.name, version = pkg.version),
+            purl = pkg.purl,
+            authors = pkg.parties.toAuthors(),
+            declaredLicenses = sortedSetOf(pkg.declaredLicense),
```

Comment:
This apparently needs to be filtered for non-empty licenses. (Or fixed in python-inspector itself.)